pkg/utils/infrastructure.go

```diff
@@ -0,0 +1,38 @@
+// utils package should be used by clients and should not import any client packages to avoid import cycles
+package utils
+
+import (
+	"os"
+	"strings"
+	// following has predefined provider identifier types, but removing dependency as it requires go 1.17
+	//configv1 "github.com/openshift/api/config/v1"
+)
+
+type Infrastructure struct {
+}
+
+func (f Infrastructure) IsExported() bool {
+	return true
+}
+
+const (
+	TYPE_AWS     = "AWS"
+	TYPE_GCP     = "GCP"
+	TYPE_INVALID = "invalid"
+)
+
+// PlatformType returns AWS if CLI input AWS profile is set/ or CLI input cloudType=AWS/ or env var AWS_ACCESS_KEY_ID or AWS_PROFILE are set
+// returns GCP if CLI input cloudType=GCP
+// returns "invalid" platformtype otherwise
+func PlatformType(cliPlatformType string, cliAwsProfile string) string {
```

Comment:
Maybe this should take a struct that wraps all the possible creds options? Passing something AWS specific here is confusing as this function should be the one figuring out the platform.